### PR TITLE
Add video predict

### PR DESCRIPTION
**Pull request by joaqo**
Makes the `predict` command accept videos. There is no change in the api, just select a video, a folder of videos, or a folder of videos and images, using the same command you'd use if they were just images.

Currently doesn't produce a json with the prediction, only creates a predicted video output. A decision has yet to be made in the team as to how said json should be formatted in the case of videos.

### CODE REVIEW

File: luminoth/predict.py [hunk 1]
```diff
@@ -1,18 +1,26 @@
+import numpy as np
+
 import click
 import json
 import os
+import skvideo.io
```
Comment (vierja) [inline luminoth/predict.py]: This needs ffmpeg installed right? Do it fail when importing or when trying to use it? Should we catch the error manually?

File: luminoth/tools/server/web.py [hunk 2]
```diff
@@ -89,7 +70,11 @@ def web(config_files, host, port, debug):
         # Gets the names of the classes
         classes_file = os.path.join(config.dataset.dir, 'classes.json')
         if tf.gfile.Exists(classes_file):
-            app.config['class_labels'] = json.load(
+            config['class_labels'] = json.load(
```
Comment (vierja) [inline luminoth/tools/server/web.py]: This is not used anymore, right?

File: luminoth/tools/server/web.py [hunk 3]
```diff
@@ -89,7 +70,11 @@ def web(config_files, host, port, debug):
...
                 tf.gfile.GFile(classes_file))
 
+    # Initialize model
+    global PREDICTOR_NETWORK
+    PREDICTOR_NETWORK = PredictorNetwork(config_files)
```
Comment (vierja) [inline luminoth/tools/server/web.py]: Do this mean that when running `lumi server web` you can't access the website until the network is fully loaded?

Comment (joaqo) [reply luminoth/tools/server/web.py]: Yep. Also separating prediction and model creation will make it cleaner to support switching between different models when we add SSD, Retina, etc.

File: luminoth/tools/server/web.py (hunk 2 above)
Comment (joaqo) [reply luminoth/tools/server/web.py]: Yep, will remove.

Comment (joaqo) [reply luminoth/tools/server/web.py]: True, will remove it.

File: luminoth/predict.py (hunk 1 above)
Comment (joaqo) [reply luminoth/predict.py]: It does fail, will catch the error

File: luminoth/predict.py [hunk 4]
```diff
@@ -19,94 +27,202 @@ def is_image(filename):
...
+    successes = 0
+    created_files_paths = []
+    total_files = len(file_paths)
+    if total_files == 0:
+        no_files_message = ("No images or videos found. "
+                            "Accepted formats -> Image: {} - Video: {}")
+        tf.logging.error(no_files_message.format(IMAGE_FORMATS, VIDEO_FORMATS))
+        exit()
+
+    # -- Initialize model --
```
Comment (nagitsu) [inline luminoth/predict.py]: Write the plain comment (e.g. `# Initialize model.`).

File: luminoth/predict.py [hunk 5]
```diff
@@ -19,94 +27,202 @@ def is_image(filename):
...
-            probs = res['objects_labels_prob']
-            object_iter = zip(objects, labels, probs)
+        save_path = 'pred_' + os.path.basename(file_path)
+        if output_dir:
+            save_path = os.path.join(output_dir, save_path)
 
-            for ind, (bbox, label, prob) in enumerate(object_iter):
-                if prob < min_prob:
+        if get_filetype(file_path) == 'image':
+            print('Predicting {}...'.format(file_path))
```
Comment (nagitsu) [inline luminoth/predict.py]: Use `click.echo` instead of `print`, here and elsewhere.

File: luminoth/tools/server/web.py [hunk 6]
```diff
@@ -1,26 +1,23 @@
...
-LOADED_MODELS = {}
-
 
 def get_image():
     image = request.files.get('image')
     if not image:
         return None
 
-    image = Image.open(image.stream)
+    image = Image.open(image.stream).convert('RGB')
```
Comment (nagitsu) [inline luminoth/tools/server/web.py]: Shouldn't the `tf.errors.OutOfRangeError` exception be captured as above?

Comment (joaqo) [reply luminoth/tools/server/web.py]: The tf.errors.OutOfRangeError seems to be caused by opening an incompatible file with tf.gfile.Open, which is not an issue on the webserver.

**Pull request merged**